src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonfwJavaProfile.java

```diff
@@ -0,0 +1,114 @@
+package com.devonfw.ide.sonarqube.common.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
+import org.sonar.plugins.java.Java;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * This class creates a quality profile containing the rules of this plugin plus additional rules from external repos.
+ */
+@SonarLintSide
+public class DevonfwJavaProfile implements BuiltInQualityProfilesDefinition {
+
+  private static final Set<String> FORBIDDEN_RULE_KEYS = new HashSet<>(Arrays.asList(
+      /* squid repo */
+      "S2076", "S2078", "S3318", "S2070", "S4142",
+      /* findbugs repo */
+      "VA_FORMAT_STRING_ARG_MISMATCH", "VA_FORMAT_STRING_MISSING_ARGUMENT", "VA_FORMAT_STRING_EXTRA_ARGUMENTS_PASSED",
+      "VA_FORMAT_STRING_BAD_CONVERSION_FROM_ARRAY", "VA_FORMAT_STRING_BAD_ARGUMENT",
+      "VA_FORMAT_STRING_BAD_CONVERSION_TO_BOOLEAN", "VA_FORMAT_STRING_NO_PREVIOUS_ARGUMENT", "VA_FORMAT_STRING_ILLEGAL",
+      "VA_FORMAT_STRING_BAD_CONVERSION", "VA_FORMAT_STRING_EXPECTED_MESSAGE_FORMAT_SUPPLIED",
+      /* fb-contrib repo */
+      "SPP_NULL_CHECK_ON_MAP_SUBSET_ACCESSOR", "SPP_NULL_CHECK_ON_OPTIONAL", "SPP_USE_CONTAINSKEY"));
+
+  private static final String DEVONFW_JAVA = "/com/devonfw/ide/sonarqube/common/rules/devon4j/devonfwJava.xml";
+
+  private static final Logger logger = Logger.getGlobal();
+
+  @Override
+  public void define(Context context) {
+
+    NewBuiltInQualityProfile devonfwJava = context.createBuiltInQualityProfile("devonfw Java", Java.KEY);
+    NodeList ruleList = readQualityProfileXml();
+    NodeList childrenOfRule;
+    NewBuiltInActiveRule currentRule;
+    String repoKey = null;
+    String ruleKey = null;
+    String severity = null;
+
+    for (int i = 0; i < ruleList.getLength(); i++) {
+
+      childrenOfRule = ruleList.item(i).getChildNodes();
```

Comment:
Looks like typical ugly DOM parsing code. Actually inside the codebase of sonarqube there must already be the logic for this import. I assume you have already searched if we can reuse this existing code and came to a negative result. If that is the case, then this should be fine. However, the XML format was defined by SonarQube and it might change. When we want to change the rules we would typically maintain them via Admin UI of SonarQube and then export them again to XML so reuse would be better. But feel free to ignore this comment if you already researched this (as I assume)...

Reply:
I did research it, but only in the sonar-java project, where the SonarQube rules and quality profiles are implemented. I did some more digging and there really is a [class for parsing xml files for quality profiles](https://github.com/SonarSource/sonarqube/blob/master/server/sonar-webserver-webapi/src/main/java/org/sonar/server/qualityprofile/QProfileParser.java), which is located in the module for their web api. Thanks for the hint!

Reply:
It seems that this QProfileParser was added for version 7.9 in September 2019. Unfortunately there is no newer .jar of the sonar-server than of version 7.2.1 which was released in June 2018.

Reply:
SonarQube used a different interface for extracting their repo & rule keys out of these generated XML files, but the logic is basically the same (reading the XML file as an input strem & if-conditions checking the key of the current XML element). Should I make further adjustments or is the parsing fine like that?